docs/how-to-guides/adding-or-reusing-tests.md

```diff
@@ -14,103 +14,198 @@ Let's inspect the test setup in `sdk/python/tests/integration`:
-│   ├── test_feature_service_read.py
 │   ├── test_online_retrieval.py
+│   ├── test_push_features_to_online_store.py
 │   └── test_universal_online.py
-├── registration
-│   ├── test_cli.py
-│   ├── test_cli_apply_duplicated_featureview_names.py
-│   ├── test_cli_chdir.py
-│   ├── test_feature_service_apply.py
-│   ├── test_feature_store.py
-│   ├── test_inference.py
-│   ├── test_registry.py
-│   ├── test_universal_odfv_feature_inference.py
-│   └── test_universal_types.py
-└── scaffolding
-    ├── test_init.py
-    ├── test_partial_apply.py
-    ├── test_repo_config.py
-    └── test_repo_operations.py
-
-8 directories, 27 files
-```
+└── registration
+    ├── test_feature_store.py
+    ├── test_inference.py
+    ├── test_registry.py
+    ├── test_sql_registry.py
+    ├── test_universal_cli.py
+    ├── test_universal_odfv_feature_inference.py
+    └── test_universal_types.py
 
-`feature_repos` has setup files for most tests in the test suite and pytest fixtures for other tests. These fixtures parametrize on different offline stores, online stores, etc. and thus abstract away store specific implementations so tests don't need to rewrite e.g. uploading dataframes to a specific store for setup.
+```
 
-## Understanding an example test
+- `feature_repos` has setup files for most tests in the test suite.
+- `conftest.py` and some of the individual test files contain fixtures which can be used to  on different offline stores, online stores, etc. and thus abstract away store specific implementations so we don't need to rewrite the same test implementation for different stores.
+
+## Structure of the test suite
+
+### What is the universal test suite?
+
+The universal test suite verifies that crucial Feast functions (e.g `get_historical_features`, `get_online_features` etc.) have the correct behavior for each of the different environments that Feast could be used in. These environments are combinations of an offline store, online store, and provider and the universal test suite serves to run basic functional verification against all of these different permutations.
+
+We use pytest [fixtures](https://docs.pytest.org/en/6.2.x/fixture.html) to accomplish this without writing excess code.
+
+Tests in Feast are split into integration and unit tests.
+
+### Is it an integration or unit test?
+
+- Integration tests test non local Feast behavior. Integration tests mainly involve testing of Feast components that connect to services outside of Feast(e.g connecting to gcp or aws clients).
+    - Generally if the test requires the initialization of a feature store in an external environment in order to test (i.e using our universal test fixtures), it is probably an integration test.
+- Unit tests, on the other hand, unit tests primarily test local and class level behavior that does not require spinning up an external service. If your test can be run locally without using any other services besides pytest, it is a unit test.
+
+### Main Types of Tests
```

Comment:
to be consistent with capitalization, this should probably be "Main types of tests"

Reply:
fixed